src/lib/components/AccessRightField.restricted.test.js

```diff
@@ -0,0 +1,247 @@
+import React from 'react';
+import { Formik } from 'formik';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { act } from 'react-dom/test-utils';
+import { AccessRightFieldCmp } from './AccessRightField';
+
+let container;
+
+beforeEach(() => {
+  // setup a DOM element as a render target
+  container = document.createElement('div');
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  // cleanup on exiting
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+// 2
+it('Can render restricted access - record with files, restricted comm', () => {
+  const setFieldValueMock = jest.fn();
+
+  const formikProps = {
+    fieldPath: 'access',
+    formik: {
+      field: { value: { record: 'restricted', files: 'restricted' } },
+      form: {
+        setFieldValue: setFieldValueMock,
+        values: { files: { enabled: true } },
+        initialValues: {
+          access: { record: 'restricted', files: 'restricted' },
+        },
+      },
+    },
+    community: { access: { visibility: 'restricted' } },
+    isMetadataOnly: false,
+    label: 'Access',
+    labelIcon: 'shield',
+  };
+
+  act(() => {
+    render(
+      <Formik
+        initialValues={formikProps.formik.form.initialValues}
+        onSubmit={() => {}}
+      >
+        <AccessRightFieldCmp {...formikProps} />
+      </Formik>,
+      container
+    );
+  });
+
+  // check restricted button active
+  expect(
+    container.querySelector(
+      '[data-testid="protection-buttons-component-restricted"]'
+    )
+  ).toHaveClass('active');
+
+  // check public button not active
+  expect(
+    container.querySelector(
+      '[data-testid="protection-buttons-component-public"]'
+    )
+  ).not.toHaveClass('active');
+
+  // check embargo checkbox disabled
+  expect(
+    container.querySelector('[data-testid="embargo-checkbox-component"]')
+  ).toHaveClass('disabled');
+
+  // check embargo option disabled
+  expect(
+    container.querySelector('[data-testid="option-list-embargo"]')
+  ).toHaveClass('disabled');
+
+  // check if message informs about restriction
+  expect(
+    container.querySelector('[data-testid="access-message"]').textContent
+  ).toContain('Restricted');
+
+  // check if files restricted
+  expect(
+    container.querySelector('[data-testid="access-files"]').textContent
+  ).toContain('The full record is restricted');
+});
+
+// 2.1
```

Comment:
leftover?

Reply:
I think I saw your explanation comment afterwards...but is in a different test file so maybe we should include it here or be sure at least that is updated? Do we really need these numbers?

Reply:
it was for me to keep my mind sane with all the use cases but indeed not needed maybe